### ISSUE TITLE
Add --plugins command-line option

### ISSUE THREAD

**Issue by GergelyKalmar** [python/mypy#11667]
**Feature**

It seems that currently the only way to make mypy load a plugin is to add it to the configuration file as `plugins = ...`. I would love if it would be possible to invoke mypy in the command line with or without a given plugin enabled. For example:

```shell
mypy --plugins=mypy_django_plugin.main,other_plugin:custom_entry_point ...
```

**Pitch**

The main benefit of this feature is that it would allow us to dynamically load plugins when invoking mypy using `mypy.api.run` or when invoking mypy using the command line. Currently it is difficult to programmatically "inject" a plugin into a mypy run, this change would make this much easier.

**Comment by sobolevn** [comment]
This might be problematic, because, for example, our plugin (mypy-django) requires extra configuration in `mypy.ini`. Loading it dynamically would just fail without this config.

**Comment by GergelyKalmar** [comment]
That is true, however, configuration can also often be injected when necessary (see for example https://github.com/typeddjango/django-stubs/issues/764#issuecomment-986283088 for one possibility with mypy-django). Also, configuration for "optional" plugins can still be kept in `pyproject.toml`, this would just allow one to choose during runtime whether to enable a given plugin or not.

**Comment by mawillcockson** [comment]
> configuration for "optional" plugins can still be kept in `pyproject.toml`

Sorry for the off-topic: I haven't been able to find how to specify mypy plugin configuration in `pyproject.toml`. Would you have some example syntax?

e.g. in `mypy.ini`

```ini
[mypy]
plugins = "example_plugin"

[example_plugin-mypy]
example_option = example_value
```

Plugins can be enabled in `pyproject.toml`:

```toml
[tool.mypy]
plugins = ["example_plugin"]
```

But I'm not sure where to put mypy plugin configuration in `pyproject.toml`.

**Comment by GergelyKalmar** [comment]
@mawillcockson That usually depends on the plugin. For example, in the case of `django-stubs` you can specify the Django settings module in `pyproject.toml` as follows:

```
[tool.django-stubs]
django_settings_module = "myproject.settings"
```

You should consult the documentation of the plugin for more information. Note that some plugins may not support `pyproject.toml`-based configuration.

**Comment by jp-larose** [comment]
Running into this issue right now where I want to specify a plugin in mypy.api.run, but it's not recognized.

**Comment by dibrinsofor** [comment]
Any work in this direction? I want to write tests with custom plugin options and this would be super helpful

**Comment by hauntsaninja** [comment]
PR welcome!
